pkg/generators/golang/clients_generator.go

```diff
@@ -890,6 +890,13 @@ func (g *ClientsGenerator) generateRequestSource(method *concepts.Method) {
 			{{ end }}
 			return
 		}
+
+		// Impersonate wraps requests on behalf of another user.
+		// Note: Only clusters service API calls support impersonation.
+		func (r *{{ $requestName }}) Impersonate(user string) *{{ $responseName }} {
```

Comment:
The return value needs to be `*{{ $requestName }}`, like the existing `Header` method. That is why the tests are failing.

Also please put this immediately after the `Header` method, here:

https://github.com/openshift-online/ocm-api-metamodel/blob/496cb39f3f7f298cf716f02e6cda9a673ed32cee/pkg/generators/golang/clients_generator.go#L798-L802

I think that is a more logical place.

Your concern about the header being ignored is very reasonable. Addressing it with a comment is also reasonable. Take into account that the clusters management service supports the header only for a subset of endpoints, not for all of them. I'd suggest to reword a bit this comment to explain in a generic way (not specific for the clusters service) that the service may not support the `Impersonate-User` header, and that if that is the case it will just be silently ignored.